static/app/utils/profiling/renderers/internalSampleTickRenderer.tsx

```diff
@@ -0,0 +1,66 @@
+import {mat3} from 'gl-matrix';
+
+import {colorChannelsToFillStyle} from '../colors/utils';
+import {Flamegraph} from '../flamegraph';
+import {FlamegraphTheme} from '../flamegraph/flamegraphTheme';
+import {getContext, Rect} from '../gl/utils';
+
+function computeIntervals(startedAt: number, weights: readonly number[]) {
+  const timeline = [startedAt + weights[0]];
+  for (let i = 1; i < weights.length; i++) {
+    timeline.push(timeline[i - 1] + weights[i]);
+  }
+  return timeline;
+}
+
+class InternalSampleTickRenderer {
+  canvas: HTMLCanvasElement;
+  context: CanvasRenderingContext2D;
+  theme: FlamegraphTheme;
+  configSpace: Rect;
+  flamegraph: Flamegraph;
+  intervals: number[];
+
+  constructor(
+    canvas: HTMLCanvasElement,
+    flamegraph: Flamegraph,
+    configSpace: Rect,
+    theme: FlamegraphTheme
+  ) {
+    this.canvas = canvas;
+    this.flamegraph = flamegraph;
+    this.theme = theme;
+    this.configSpace = configSpace;
+    this.intervals = computeIntervals(
+      this.configSpace.x,
+      this.flamegraph.profile.weights
+    );
+    this.context = getContext(canvas, '2d');
+  }
+
+  draw(
+    configViewToPhysicalSpace: mat3,
+    context: CanvasRenderingContext2D = this.context
+  ): void {
+    if (this.intervals.length === 0) {
+      return;
+    }
+    const height =
+      this.theme.SIZES.LABEL_FONT_SIZE * window.devicePixelRatio +
+      this.theme.SIZES.LABEL_FONT_PADDING * window.devicePixelRatio * 2 -
+      this.theme.SIZES.LABEL_FONT_PADDING;
+    context.strokeStyle = colorChannelsToFillStyle(this.theme.COLORS.RED);
+    context.lineWidth = this.theme.SIZES.INTERNAL_SAMPLE_TICK_LINE_WIDTH;
+
+    for (let i = 0; i < this.intervals.length; i++) {
```

Comment:
Yes, but it means you still need to compute the final canvas space coordinates, you can skip that if you compare to the view

Reply:
i need to change my mental model here :) ... i'm naturally leaning toward actual values in 'physical' space instead of their scaled counterparts that derive physical space.

in this case i'm assuming you mean, `flamegraphView.configView` ?